proposals/yaml_signing_and_verification.md

```diff
@@ -0,0 +1,168 @@
+# YAML Signing & Verification 
+
+Name: YAML signing and verification \
+Start Date: 2021-12-25 \
+Authors: [Namanl2001](https://github.com/Namanl2001), [JimBugwadia](https://github.com/JimBugwadia)\
+Supersedes: N/A
+
+## Overview
+
+This proposal adds support for verifying the integrity and source of a Kubernetes resource using cosign signatures and a Kyverno policy.
+
+## Motivation
+
+For certain Kubernetes resources, like central policies applied across multiple clusters, it is important to ensure that the resources can be trusted, and that any attempts to tamper with the resource  definition is blocked.  
+
+Signatures are commonly used to verify origin and content of critical resources. For example, container image signing is used to verify the source and integrity of images.
+
+This feature uses signatures to sign resource declarations via Cosign, and verify the integrity of the resource using Kyverno.
+
+## Proposal
+
+The proposal is to support the sigstore [k8s-manifest CLI](https://github.com/sigstore/k8s-manifest-sigstore)  for signing YAMLs using the option to embed the signature into the YAML declaration:
+
+Here is an example of signing an nginx pod using a static key:
+
+```
+ kubectl sigstore sign -f nginx-pod.yaml  --key cosign.key
+```
+
+This produces a `nginx-pod.yaml.signed` declaration that looks like this:
+
+```
+ apiVersion: v1
+kind: Pod
+metadata:
+  annotations:
+    cosign.sigstore.dev/message: H4sIAAAAAAAA/wAzAcz+H4sIAAAAAAAA/+ySPU/DMBCGM+dXnLqH2v2IW6/sKANiP5IjMjg+y3YLBfHfUdwKmEB0qZDyLCf7PT/+kOdp8POWBx8oRuP6KmGo+tdlLWW9larezl1v3Evlubs64GCLMxBCCLVej1WqtfheM0u1KuRK1lJJpepFIRZCrZYFiHM2+yu7mDAUQjya4ce+3/LTXT7rPwG9uaMQDTsNe1k+GddpaLgrB0rYYUJdAqBznDAZdlHD23sJYPGebBwzALSWn6uOLCWKGmYPaCPNSgCHA2nw3FX5D5XRUzsuadklNI5CFlRgBuxJQ27SFhPFlMV5vtlZ27A17UHDDe0p5OioPmrHseeQTsepvvwNh6RhIzYiJwA+cOKWrYbb66a89NtPTExMXJKPAAAA//+lixZxAAgAAAEAAP//QYqwEDMBAAA=
+    cosign.sigstore.dev/signature: MEQCIGJ9cjguB6TXcMGCDyWyCRyVOb6YOrHDo6S4VtZGuvVjAiA1qm5N/koj94I4JtL5BywhuBVxS7VpEkGLtcwIkVDogg==
+  labels:
+    allow-deletes: "false"
+  name: pod-nginx
+spec:
+  containers:
+  - image: nginx:latest
+    imagePullPolicy: Never
+    name: nginx
+    ports:
+    - containerPort: 8080
+      protocol: TCP
+
+```
+
+This declaration can then be verified by a Kyverno policy rule. This policy checks that all Kyverno policies are signed using a static key:
```

Comment:
Since the previous example showed signing a Pod manifest, please show a (Cluster)Policy that demonstrates how to verify the Pod manifest, not a Kyverno resource.